packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -464,6 +472,66 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.sendSubscriptionPaymentFailedEmail(invoice);
   }
 
+  /**
+   * Handle `invoice.upcoming` Stripe wehbook events.
+   */
+  async handleInvoiceUpcomingEvent(request: AuthRequest, event: Stripe.Event) {
+    const invoice = event.data.object as Stripe.Invoice;
+    const customer = await this.stripeHelper.expandResource(
+      invoice.customer,
+      CUSTOMER_RESOURCE
+    );
+
+    if (
+      customer?.deleted ||
+      !customer?.invoice_settings?.default_payment_method
+    ) {
+      return;
+    }
+
+    const { card } = await this.stripeHelper.expandResource(
+      customer?.invoice_settings.default_payment_method,
+      PAYMENT_METHOD_RESOURCE
+    );
+
+    if (card?.exp_month && card.exp_year) {
+      // If card expiry is greater than current date, return without further action.
```

Comment:
I think CCs expire at the end of the month, so this comment is a little misleading.